pkg/reconciler/pipelinerun/resources/pipelinerunresolution.go

```diff
@@ -580,6 +581,52 @@ func GetTaskRunName(taskRunsStatus map[string]*v1beta1.PipelineRunTaskRunStatus,
 	return kmeta.ChildName(prName, fmt.Sprintf("-%s", ptName))
 }
 
+// GetNamesofTaskRuns should return unique names for `TaskRuns` if one has not already been defined, and the existing one otherwise.
+func GetNamesofTaskRuns(taskRunsStatus map[string]*v1beta1.PipelineRunTaskRunStatus, childRefs []v1beta1.ChildStatusReference, ptName, prName string, combinationCount int) []string {
+	var taskRunNames []string
+
+	taskRunNames = getTaskRunNamesFromChildRefs(childRefs, ptName)
+	if taskRunNames != nil {
+		return taskRunNames
+	}
+
+	taskRunNames = getTaskRunNamesFromTaskRunsStatus(taskRunsStatus, ptName)
+	if taskRunNames != nil {
+		return taskRunNames
+	}
+
+	return getNewTaskRunNames(ptName, prName, combinationCount)
+}
+
+func getTaskRunNamesFromChildRefs(childRefs []v1beta1.ChildStatusReference, ptName string) []string {
+	var taskRunNames []string
+	for _, cr := range childRefs {
+		if cr.Kind == pipeline.TaskRunControllerName && cr.PipelineTaskName == ptName {
+			taskRunNames = append(taskRunNames, cr.Name)
+		}
+	}
+	return taskRunNames
+}
+
+func getTaskRunNamesFromTaskRunsStatus(taskRunsStatus map[string]*v1beta1.PipelineRunTaskRunStatus, ptName string) []string {
+	var taskRunNames []string
+	for k, v := range taskRunsStatus {
+		if v.PipelineTaskName == ptName {
+			taskRunNames = append(taskRunNames, k)
+		}
+	}
+	return taskRunNames
+}
+
+func getNewTaskRunNames(ptName, prName string, combinationCount int) []string {
+	taskRunName := kmeta.ChildName(prName, fmt.Sprintf("-%s", ptName))
+	var taskRunNames []string
+	for i := 0; i < combinationCount; i++ {
+		taskRunNames = append(taskRunNames, fmt.Sprintf("%s-%d", taskRunName, i))
```

Comment:
This increases the length of the name by at least two more characters (`-` and matrix id). The matrix id could be in 1000...s (hopefully not). But at the least (for 10 count), this results in the name with 65 characters instead of 63 (the current limit).

Can we include the combination count while calling `ChildName`:

```
kmeta.ChildName(prName, fmt.Sprintf("-%s-%d", ptName,i))
```

This will result in a separate suffix all the time. 

The other alternative is to re-write `ChildName` for our purpose which is fine as well.

Reply:
@pritidesai - good catch, this is fixed now - and it's okay to have a different suffix, because we have the matrixId

https://github.com/tektoncd/community/blob/main/teps/0090-matrix.md#combinations-of-parameters-in-the-matrix

Reply:
that's a good point @jerop Is this `taskRun` part of minimal pipelineRun status? I am guessing its not which is fine since `taskRun` has `pipelineTask` name. So, the users can classify `taskRun`s belonging to one single `pipelineTask` based on the owner reference.

I am not sure how much matrixId is valuable here in the `taskRun` status since it's a number determined by the tekton controller which cannot be directly mapped to matrix params or a combination of matrix params.

Reply:
@pierretasci supporting results from fanned out tasks is out of scope for TEP-0090 - https://github.com/tektoncd/community/blob/main/teps/0090-matrix.md#results-from-fanned-out-pipelinetasks

results are referenced using the `PipelineTask` names - but in matrixed cases, there are multiple `TaskRuns` - so the matrixId could be useful in gathering the results from matrixed pipelinetasks, such as if we use an object result 

open to updating the TEP to remove that part so that the complete problem will be addressed in the TEP focusing on supporting results from matrixed pipelinetasks